### PR TITLE
Improving Earn button and adding tracking events

### PR DESCRIPTION
**Pull request by cmonti-bc**
## Description (optional)
* Adding a conditional for the button to be hidden while user or coin is not elegible or not golden tier for using rewards.
* Adding tracking events for the buy, sell and earn buttons.

### Demo (desktop view)
TrmhsJfDtl

### Demo (mobile view)
gVjDyqzOOv

### Demo (event tracking)
goyUsuTW6f

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/analytics/types/transactions.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export enum Events {
+  TRANSACTIONS_EARN_BUTTON_CLICKED = 'Transactions Earn Button Clicked'
```
Comment (cmonti-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/analytics/types/transactions.ts:2]: Adding transaction earn event only as we still have [the legacy ones for sell & buy](https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/6db5e21daa04378b4715ea5eda3871239991ea95/packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts#L12).

Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/analytics/types/transactions.ts:2]: from which library from segment you get this? usually we reflect segment structure to this place for easier later maintenance, also once when new types file is about to be added it should contains all existing events :)

**Review by milan-bc — CHANGES_REQUESTED** [review]
great progress so far @cmonti-bc, let's align segment library with namings

File: packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/index.tsx [hunk 2]
```diff
@@ -141,7 +148,10 @@ class TransactionsContainer extends React.PureComponent<Props> {
     } = this.props
     const { coin } = computedMatch.params
     const { coinfig } = window.coins[coin]
-    const interestEligibleCoin = interestEligible[coin] && interestEligible[coin]?.eligible
+    const interestEligibleCoin =
+      interestEligible && interestEligible[coin] && interestEligible[coin]?.eligible
+    const isEarnButtonEnabled = isGoldTier && interestEligibleCoin
+    const buttonResponsiveWidth = isMobile() ? 'auto' : '100px'
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/index.tsx:154]: please use a styled component for CSS rules

File: packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/selectors.ts [hunk 3]
```diff
@@ -126,10 +125,20 @@ export const getData = (state: RootState, ownProps: OwnProps) => {
...
+      invitationsR,
+      userSearch,
+      pagesR,
+      currencyR,
+      recurringBuys,
+      isRecurringBuyR,
+      interestEligibleResult
+    ) => {
       const empty = (page) => isEmpty(page.data)
+      const interestEligible = propOr(false, 'data', interestEligibleResult)
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/selectors.ts:141]: you need to map over the Remote using one of its instance methods or use a custom to get the value. never access `data` directly

**Review by schnogz — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/analytics/types/transactions.ts (hunk 1 above)
Comment (cmonti-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/analytics/types/transactions.ts:2]: Thanks for the guidance @milan-bc I'll fix this.

Comment (cmonti-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/analytics/types/transactions.ts]: This is now fixed, please review @milan-bc

File: packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/selectors.ts [hunk 4]
```diff
@@ -126,10 +125,20 @@ export const getData = (state: RootState, ownProps: OwnProps) => {
...
     ],
-    (invitationsR, userSearch, pagesR, currencyR, recurringBuys, isRecurringBuyR) => {
+    (
+      invitationsR,
+      userSearch,
+      pagesR,
+      currencyR,
+      recurringBuys,
+      isRecurringBuyR,
+      interestEligibleResult
```
[the remark this answers is not in the document]
Comment (cmonti-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/selectors.ts]: I guess this is now fixed.

File: packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/index.tsx (hunk 2 above)
Comment (cmonti-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/index.tsx]: This is fixed. Thanks

File: packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/selectors.ts (hunk 3 above)
Comment (cmonti-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/selectors.ts]: I think this is fixed now. Please review @schnogz when you can.

File: packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/index.tsx [hunk 5]
```diff
@@ -328,6 +358,7 @@ const mapDispatchToProps = (dispatch: Dispatch, ownProps: OwnProps) => {
   const { coin } = ownProps.computedMatch.params
   const { coinfig } = window.coins[coin]
   const baseActions = {
+    analyticsActions: bindActionCreators(actions.analytics, dispatch),
```
Comment (wford-bc) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/index.tsx:361]: Testing a comment

**Review by wford-bc — CHANGES_REQUESTED** [review]
Just testing

File: packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/index.tsx (hunk 5 above)
Comment (cmonti-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Transactions/index.tsx:361]: Answer test

**Review by wford-bc — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/analytics/types/coinView.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+export enum Events {
+  COINVIEW_EARN_REWARDS_BUTTON_CLICKED = 'Coin View Earn Rewards Clicked'
+}
+
+type CoinViewEarnRewardsButtonClick = {
+  key: Events.COINVIEW_EARN_REWARDS_BUTTON_CLICKED
+  properties: {
+    coin: string
```
Comment (wford-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/analytics/types/coinView.ts:8]: Should this be currency?

Comment (wford-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/analytics/types/coinView.ts:8]: And what about device and platform?

**Review by wford-bc — DISMISSED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/analytics/types/coinView.ts (hunk 6 above)
Comment (cmonti-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/analytics/types/coinView.ts]: Device and platform added for the tracking of the event. Please review.

**Review by wford-bc — APPROVED** [review]

**Review by schnogz — APPROVED** [review]

**Review by julianromero-bc — APPROVED** [review]

**Pull request merged**